src/Publisher/AbstractCloudPublisher.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Lmc\Steward\Publisher;
+
+use Facebook\WebDriver\Remote\RemoteWebDriver;
+use Lmc\Steward\Test\AbstractTestCaseBase;
+
+/**
+ * Abstract publisher for cloud services with HTTP API to store test results
+ */
+abstract class AbstractCloudPublisher extends AbstractPublisher
+{
+    /** @var string Content type of the request */
+    const CONTENT_TYPE = 'application/x-www-form-urlencoded';
+
+    public function publishResults(
+        $testCaseName,
+        $status,
+        $result = null,
+        \DateTimeInterface $startDate = null,
+        \DateTimeInterface $endDate = null
+    ) {
+        // we publish only result for each separate test using publishResult()
```

Comment:
This method is not called by user, it is called from the Steward (TestStatusListener) and the method it is called on all registered listeners.